unittests/test_cli.py

```diff
@@ -578,11 +578,11 @@ def test_filtering_multiple_criteria(run_reframe):
     returncode, stdout, stderr = run_reframe(
         checkpath=['unittests/resources/checks'],
         action='list',
-        more_options=['-t', 'foo', '-n', 'HelloTest']
+        more_options=['-t', 'foo', '-n', 'HelloTest', '-n', 'HelloMakeTest']
```

Comment:
I don't think that this change tests what we want to test. We should test a combination of `-n Test` and `-n TestX@0` options.